identity_storage/src/error.rs

```diff
@@ -0,0 +1,155 @@
+// Copyright 2020-2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::borrow::Cow;
+use std::error::Error;
+use std::fmt::Debug;
+use std::fmt::Display;
+
+/// The error type for key storage operations.
+///
+/// Instances always carry a corresponding [`StorageErrorKind`] and may be extended with custom error messages and
+/// source.
+#[derive(Debug)]
+pub struct StorageError<T: StorageErrorKind> {
```

Comment:
Using generics here is a good idea! 

It might even make sense to rename this to `CustomError<T>` and move it to `identity_core` so it can be used for all errors users need to return in their code. Not a requirement for this PR.